src/Chain.ts

```diff
@@ -115,49 +115,49 @@ export function bind<M extends URIS4>(
   M: Chain4<M>
 ): <N extends string, A, S, R, E, B>(
   name: Exclude<N, keyof A>,
-  f: (a: A) => Kind4<M, S, R, E, B>
+  f: <A2 extends A>(a: A | A2) => Kind4<M, S, R, E, B>
```

Comment:
if `A2 extends A` isn't `A | A2` redundant?

```diff
-f: <A2 extends A>(a: A | A2) => Kind4<M, S, R, E, B>
+f: <A2 extends A>(a: A2) => Kind4<M, S, R, E, B>
```

Reply:
Thank you for the feedback @gcanti.  I tried removing A according to your suggestion here, and I got the following error in line 633 of `test/TaskEither.ts`.

```
Argument of type '<E>(i: A2) => TaskEither<E, number>' is not assignable to parameter of type '<A2 extends { readonly a: number; readonly b: string; readonly c: number; readonly d: string; }>(a: A2) => TaskEither<unknown, number>'.
  Types of parameters 'i' and 'a' are incompatible.
    Type 'A2' is not assignable to type 'A2'. Two different types with this name exist, but they are unrelated.
      'A2' is assignable to the constraint of type 'A2', but 'A2' could be instantiated with a different subtype of constraint '{ readonly a: number; readonly b: string; readonly c: number; readonly d: string; }'
```

I also tried `<A2 extends A = A>(a: A2) => Kind4<M, S, R, E, B>`, but got the same error.  I unfortunately can't claim to understand the nuance here.

Reply:
I played around with your suggestion a bit more.  I changed the test to use `fromOptionK` instead of `fromIOK`, and it works  with your suggested changes as expected now.  I'm not sure I understand the signature for fromIOK.

I've updated all occurrences of `A | A2` to be just `A2`.

Reply:
> I changed the test to use fromOptionK instead of fromIOK

If you remove the type annotation here

```diff
-_.bind('e', _.fromOptionK(() => 'err')((p: { readonly c: number }) => some(p.c)))
+_.bind('e', _.fromOptionK(() => 'err')((p) => some(p.c)))
```

`fromOptionK` raises an error too. 

Looks like I was wrong and `A | A2` is necessary, though the reason why is not clear to me

Reply:
Thank you for the follow-up!  I changed the type signature back to use `A | A2`, separated out the param passthrough test, and added a case to provide coverage for `bindW`